neutrons/histogram-dev
# Pre commit setup

Created .pre-commit-config.yaml file. Ignored a few errors. It's better to correct these errors after I can install and run tests to prevent accidentally breaking something.

To install: (git checkout -b Pre-commit-Setup origin/Pre-commit-Setup)
1. set up conda environment "conda env create" using environment.yml file.
2. conda activate histogram-dev
3. in terminal, go into python/ locate setup.py. Run pip install -e. in this terminal

To test:
cd ~/histogram-dev/tests/histogram
run alltests.py either in VS code or terminal: python alltests.py. All tests should pass in tfile.

cd ~/historgram-dev/tests/histogram/ndarray
run alltests.py either in VS code or terminal: python alltests.py. Expect run 28 tests, 5 would fail.


Failing tests:
1.ERROR: testNumpyNdArray2StdVectorNdArray (converters_TestCase.converters_TestCase)
histogram.ndarray.converters: NumpyNdArray2StdVectorNdArray

2.ERROR: testStdVectorNdArray2NumpyNdArray (converters_TestCase.converters_TestCase)
histogram.ndarray.converters: StdVectorNdArray2NumpyNdArray

3.ERROR: test__div__ (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: operator 'a/b'

4.ERROR: test_dump (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: pickle.dump

5.ERROR: test_load (NdArray_TestCase.createTestCase.<locals>.TC)
NdArray: pickle.load


(Currently next(default), main, qa are the same as danse-inelastic/histogram:master)
